slides/12_Remix.html

```diff
@@ -24,6 +24,727 @@
+    ├── blog.archive.tsx
+    └── blog.authors.tsx
+              </script></code></pre>
+            <p>
+              &rarr; e.g.
+              <span class="code">/blog/five-great-tips-for-2024</span>
+            </p>
+          </section>
+          <section>
+            See
+            <a
+              href="https://remix.run/docs/en/1.19.3/file-conventions/route-files-v2"
+              >Remix docs</a
+            >
+            for more details.
+          </section>
+        </section>
+        <section>
+          <section data-auto-animate>
+            <h3>Route Module API</h3>
+          </section>
+          <section data-auto-animate>
+            <h3>Route Module API</h3>
+
+            <pre><code class="hljs no-highlight" data-trim data-line-numbers="6" data-id="code"><script type="text/template">
+              app/routes/
+                  ├── _index.tsx
+                  ├── about.tsx
+                  ├── pricing.tsx
+                  ├── blog._index.tsx
+                  ├── blog.$slug.tsx
+                  ├── blog.archive.tsx
+                  └── blog.authors.tsx
+                            </script></code></pre>
+          </section>
+          <section>
+            <p><span class="code">blog.$slug.tsx</span></p>
+            <pre><code class="hljs tsx" data-trim data-line-numbers="|3,10|5-9|1,4" data-id="code"><script type="text/template">
+          import { useParams } from "@remix-run/react";
+
+          export default function BlogPostPage() {
+            const { slug } = useParams();
+            return (
+              <p>
+                This is the blog post with slug {slug}.
+              </p>
+            );
+          }
+        </script></code></pre>
+            <aside class="notes">
+              <ol>
+                <li>Must be default export</li>
+                <li>It's a React component, return JSX</li>
+                <li>
+                  useParams like in React Router (Disclaimer: just as an example
+                  – Data fetching will work differently (later)
+                </li>
+              </ol>
+            </aside>
+          </section>
+          <section data-auto-animate>
+            <p>How can we redirect to a different route?</p>
+
+            <p class="fragment">
+              E.g. <span class="code">/</span> &rarr;
+              <span class="code">/blog</span>
+            </p>
+          </section>
+          <section data-auto-animate>
+            <p>How can we redirect to a different route?</p>
+            <p>
+              E.g. <span class="code">/</span> &rarr;
+              <span class="code">/blog</span>
+            </p>
+
+            <div>
+              <p><span class="code">_index.tsx</span></p>
+              <pre><code class="hljs tsx" data-trim data-line-numbers="3,5|1,4|" data-id="code"><script type="text/template">
+import { redirect } from "@remix-run/node";
+
+export function loader() {
+  return redirect("blog");
```

Comment:
**Suggestion:** Add a hint mentioning, that this will be a pure serverside redirect, meaning the corresponding http return type will be sent to the client with the new redirect target.
For a client side navigation (i.e. clicking some non-link component) is analog to react-router with useNavigate.

Reply:
Agree, that's a good addition to make it clearer. It would also apply to Link components though (not sure if I get you right there in the last sentence). Only for a plain a-href it would perform a full page navigation.